lib/phx_tools_web/live/phx_tools_live/index.ex

```diff
@@ -2,31 +2,49 @@ defmodule PhxToolsWeb.PhxToolsLive.Index do
   use PhxToolsWeb, :live_view
 
   alias PhxToolsWeb.Endpoint
-  alias PhxToolsWeb.InstructionComponents
-  alias PhxToolsWeb.PhxToolsLive.LandingComponent
+  alias PhxToolsWeb.PhxToolsComponents
 
   @impl Phoenix.LiveView
   def mount(_params, session, socket) do
     {:ok,
      socket
      |> assign(:seo_attributes, %{})
-     |> assign_os(session)}
+     |> assign_os_and_source_code_url(session)}
   end
 
-  defp assign_os(socket, %{"operating_system" => operating_system}),
-    do: assign(socket, :operating_system, operating_system)
+  defp assign_os_and_source_code_url(socket, %{"operating_system" => operating_system}) do
+    socket
+    |> assign(:operating_system, operating_system)
+    |> assign(
+      :source_code_url,
+      "https://github.com/optimumBA/phx.tools/blob/main/priv/static/#{operating_system}.sh"
```

Comment:
I see this URL in two places.

Reply:
I decided to assign it dynamically on mount using the detected OS, but every time an action is applied I have to update it using the applied action.

Reply:
Can't you have this in some private function that's called from both places or something?